spark/src/test/scala/org/apache/spark/sql/delta/optimize/OptimizeZOrderSuite.scala

```diff
@@ -193,6 +194,94 @@ trait OptimizeZOrderSuiteBase extends OptimizePartitionTableHelper
     }
   }
 
+  def optimizeWithBatching(
```

Comment:
Is this function same as the one in spark/src/test/scala/org/apache/spark/sql/delta/optimize/OptimizeCompactionSuite.scala? Can we refactor to share the implemnetation?

Reply:
They're the same except for the `executeOptimizePath` call which has a z-order col here. I could try to refactor into a shared function or base class, but it would be a little tricky having to use the different `executeOptimizePath` calls, and the two suites don't currently share any other base optimize related base class. Might just be overkill, but I can try if you really think it's worth it.

Reply:
Made https://github.com/delta-io/delta/issues/3249